aws_topology/stackstate_checks/aws_topology/resources/cloudformation.py

```diff
@@ -65,41 +73,66 @@ def no_arn(region=None, account_id=None, resource_id=None, **kwargs):
+    def collect_stack_resources(self, stack_id):
+        return self.client.describe_stack_resources(StackName=stack_id).get("StackResources", [])
+
+    def collect_stack(self, stack_data):
+        resources = self.collect_stack_resources(stack_data.get("StackId", ""))
+        return StackData(stack=stack_data, resources=resources)
+
+    def collect_stacks(self):
+        for stack in [
+            self.collect_stack(stack_data)
+            for stack_data in client_array_operation(self.client, "describe_stacks", "Stacks")
+        ]:
+            yield stack
+
+    @set_required_access_v2("cloudformation:DescribeStacks")
+    def process_stacks(self):
+        for stack_data in self.collect_stacks():
+            self.process_stack(stack_data)
+
     def process_all(self, filter=None):
-        for stack_description_raw in self.client.describe_stacks().get("Stacks") or []:
-            stack_description = make_valid_data(stack_description_raw)
-            stack_id = stack_description["StackId"]
-            stack_name = stack_description["StackName"]
-            self.process_stack(stack_id, stack_name, stack_description)
-        for stack_data_page in self.client.get_paginator("list_stacks").paginate():
-            for stack_raw in stack_data_page.get("StackSummaries") or []:
-                stack = make_valid_data(stack_raw)
-                stack_id = stack["StackId"]
-                if "ParentId" not in stack:
-                    continue
-                parent_id = stack["ParentId"]
-                self.emit_relation(stack_id, parent_id, "child of", stack)
-
-    def process_stack(self, stack_id, stack_name, stack_description):
-        self.emit_component(stack_id, self.COMPONENT_TYPE, stack_description)
-        self.process_resources(stack_id, stack_name)
-
-    def process_resources(self, stack_id, stack_name):
-        # TODO StackName can also be sent stack_id
-        resources = self.client.describe_stack_resources(StackName=stack_name).get("StackResources") or []
-        for resource in resources:
-            self.process_resource(stack_id, resource)
+        if not filter or "stacks" in filter:
+            self.process_stacks()
+
+    @transformation()
+    def process_stack(self, data):
+        stack = Stack(data.stack, strict=False)
+        stack.validate()
+        output = make_valid_data(data.stack)
+        output["Name"] = stack.StackName
+        self.emit_component(stack.StackId, self.COMPONENT_TYPE, output)
 
-    def process_resource(self, stack_id, resource):
-        resource_type = resource["ResourceType"]
-        if resource.get("PhysicalResourceId"):
-            arn = self.agent.create_arn(resource_type, self.location_info, resource.get("PhysicalResourceId"))
-            self.emit_relation(stack_id, arn, "has resource", {})
+        resources = [StackResource(resource, strict=False) for resource in data.resources]
+        for resource in resources:
+            resource.validate()
```

Comment:
If one `resource` fails, we will stop processing the rest. Is this the desired outcome here?

Reply:
Not "desired" per se, but if we come across an API response we don't expect, it would probably not be a good idea to continue. We also exit out the same way on a per-resource basis, at the top of this function